### DISCUSSION TITLE
App Open Ads Admob Integration

### DISCUSSION THREAD

**Discussion by tomLadder** [invertase/react-native-firebase#4261] [Ideas] [answered]
Google has just sent the new ad format "App Open Ads" into open beta. Are there any plans for rnfb to support this type of ad? https://developers.google.com/admob/ios/app-open-ads

**Comment by mikehardy** [comment, marked as answer]
There are no plans at current - and more importantly, it's important to understand the Ads are actually not a part of the Firebase surface area at all any more - see this: https://github.com/FirebaseExtended/flutterfire/pull/3347#issuecomment-680182389

The philosophically correct thing to do (similar to the Notifications excision after RNFB v5) is to peel AdMob out of the library completely, and either open PRs to existing standalone libraries (https://github.com/sbugert/react-native-admob ?) or make a separate package

Which is to say, it exists now and if someone made a high-quality PR to implement new functionality or fix bugs we would certainly consider it (happily!) but there are no plans at current

**Reply by tomLadder** [reply]
Thanks @mikehardy for the fast response. I can definitely see your point.

**Reply by mikehardy** [reply]
I wanted to make sure I wasn't just being hasty here so I read the firebase-admob current docs: https://firebase.google.com/docs/admob/

Look especially at the steps, and the code. It appears to me like the only actual connection - code or otherwise - between firebase and admob now is that *if* you use firebase analytics (which enables audience definition, A/B test, some remote config stuff, plus obviously analytics) *and* you "connect your admob account with your firebase app" (in the web console) then you get connected analytics reports and bigquery stuff.

Which has nothing to actually do with implementation or APIs, so I am pretty sure my statement ("there is no API surface area here, a 3rd party library is the proper focus") stands

But we really do look at any PRs and I merge most, so if you are motivated, go for it.
